src/app/rosetta/lib/block.ml

```diff
@@ -312,9 +339,24 @@ WITH RECURSIVE chain AS (
     let run_by_id (module Conn : Caqti_async.CONNECTION) id =
       Conn.find_opt query_by_id id
 
+    let run_is_old_height (module Conn : Caqti_async.CONNECTION) ~height =
+      let open Deferred.Result.Let_syntax in
+      let open Int64 in
+      let%map max_height =
+        Conn.find query_max_height ()
+      in
+      (* buffer an epsilon of 5 just in case archive node isn't caught up *)
```

Comment:
Is the criterion to distinguish old from new queries based just on the current height, or should it examine whether blocks have marked as canonical?

Reply:
yes I was thinking we should just rely on height here -- we know performing the long recursive queries in rosetta cause Postgres to choke, so I'd rather the queries fail in the case that we haven't marked things as canonical as opposed to trying the big recursive queries in that case and then fail.